src/library.js

```diff
@@ -56,6 +56,20 @@ LibraryManager.library = {
     pgid: 42
   },
 
+  // ==========================================================================
+  // getTempRet0/setTempRet0: scratch space handling i64 return
+  // ==========================================================================
+
+  getTempRet0__sig: 'i',
+  getTempRet0: function() {
```

Comment:
these functions are defined in `support.js`, I don't think we need them in library.js, so this part can be removed. (These might be needed if we used the library deps system for them, but they are just normal support JS funcs.)

Reply:
It seems that if we want to make a symbol available to native user code it needs to be listed as a library fuctions.

Without that if there are direct references in the native code I get this:

```
error: undefined symbol: getTempRet0
warning: To disable errors for undefined symbols use `-s ERROR_ON_UNDEFINED_SYMBOLS=0`
```

I'm just getting a handle on the whole library system but it seems that directly calling.

Actually the above error only occurs with the wasm backend (I guess because with asm2wasm these symbols maybe don't show up as direct asm.js dependencies but get injected at the asm2wasm phase?  (which happens after symbols are checked).

Reply:
Sorry, yeah, now I see what you mean. Yes, JS in support.js is just copied into the output, and the linking code is not aware of it. So it does indeed need to be in library.js.

So I think we can remove those from support.js, keep them in library.js, and to make sure they are included by default, add them to `DEFAULT_LIBRARY_FUNCS_TO_INCLUDE` (although, adding them in emscripten.py - around where `DEFAULT_LIBRARY_FUNCS_TO_INCLUDE` is used perhaps - may be safer, as users  can override `DEFAULT_LIBRARY_FUNCS_TO_INCLUDE`).

Reply:
Do you might if we do that as a followup, once we land all the 4 different changes here?  I tried to move them out of support but ran into issues.

Reply:
Sounds fine as a followup, sure (assuming all tests pass with the current approach - seems like the duplication could also cause issues, but if it works for now that's cool).